### PR TITLE
Aadu/logging

### PR DESCRIPTION
**Pull request by beardedtrogon**
probably fucked something up. Lmk what to fix.

### CODE REVIEW

File: scratch/README.md [hunk 1]
```diff
@@ -0,0 +1 @@
+This is the scratchpad folder, useful for keeping track of things not directly related to the final deliverable.
```
Comment (zebra6) [inline scratch/README.md]: Add the correct entries to the gitignore and remove the files in the scratch directory that were added (fd/vadc/*)

Comment (zebra6) [reply scratch/README.md]: Actually the whole scratch folder needs to be removed (probably from an earlier commit) along with all the __pycache__ stuff.

Comment (zebra6) [reply scratch/README.md]: _ _ pycache _ _ ... double underline is bold apparently.

File: zebra_claw/example_adc_read.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python3
+import random
+import time
+import threading
+import argparse
+
+adc_dirname = './virtual/'
```
Comment (zebra6) [inline zebra_claw/example_adc_read.py]: There are standardized paths now in the util directory.  Look at the vadc to get an idea of how to use them.  Or tap tap...

File: zebra_claw/logging.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+import time
+import argparse
+import threading
+from datetime import datetime
+from threading import Timer
+
+adc_dirname = './virtual/'
```
Comment (zebra6) [inline zebra_claw/logging.py]: Same as above on paths.

File: zebra_claw/example_adc_read.py (hunk 2 above)
Comment (beardedtrogon) [reply zebra_claw/example_adc_read.py]: hah. tap tap...

File: scratch/README.md (hunk 1 above)
Comment (beardedtrogon) [reply scratch/README.md]: so im deleting _ _ pycache _ _  and the scratch folder?

**Comment by beardedtrogon** [comment]
@zebra6 take a look at the edits. Will work on the delta stuff once we have non-random numbers. Still getting issues with empty values if the ADC process is updating the file descriptor while the logger is trying to read it

**Comment by zebra6** [comment]
I'll look at it tomorrow, we probably need to work out file locking.  One thing I am noticing that needs to be investigated, is if you run the vadc the entire system will lock up every ~45 seconds or so for a second or two.  You can verify this by running `tail -f virtual/vadc0` during a run and watching the output.  Figuring out what is causing that, and detecting it in the field will be very helpful (I think it is a system thing, not the code we are running).

File: zebra_claw/logging.py [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+sys.path.append("../util")
+
+import log                      #logger
+import paths                    #global paths
+from log import g               #shorter logging commands
+from log import l               #   "
+from log import ll              #   "
+
+NUM_LOAD_CELLS = 4
+current_raw_file = ''
```
Comment (zebra6) [inline zebra_claw/logging.py]: prefix globals with g_

File: zebra_claw/logging.py [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+    parser.add_argument("-p", "--period",
+        help = "define ADC sampling period in seconds (float)",
+        action = "store",
+        default = 0.1,
+        metavar='',
+        type = float)
+
+    args = parser.parse_args()
+
+    if args.verbose:
```
Comment (zebra6) [inline zebra_claw/logging.py]: delete this

**Comment by beardedtrogon** [comment]
Take another look. Should have addressed your concerns

**Pull request merged**